src/evict/evict_page.c

```diff
@@ -324,6 +326,15 @@ __evict_review(
 	/* Now that we have exclusive access, review the page. */
 	page = ref->page;
 
+	modified = __wt_page_is_modified(ref->page);
+
+	/*
+	 * Clean pages can't be evicted when running in memory only. This
+	 * should be uncommon - we don't add clean pages to the queue.
+	 */
+	if (F_ISSET(S2C(session), WT_CONN_IN_MEMORY) && !modified)
+		return (EBUSY);
+
```

Comment:
@agorrod, there's another call to `__wt_page_is_modified` about 10 lines farther on that could be replaced with `modified`.